controllers/grid/queries/QueriesAccessHelper.inc.php

```diff
@@ -99,18 +89,18 @@ function getCanCreate() {
 	 * @return boolean True iff the user is allowed to edit the query.
 	 */
 	function getCanEdit($queryId) {
-		$userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
-
-		// Managers and sub editors are always allowed
-		if (count(array_intersect($userRoles, array(ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR)))) return true;
+		$queryDao = DAORegistry::getDAO('QueryDAO');
+		$query = $queryDao->getById($queryId);
+		if (!$query) return false;
 
-		// Assistants and authors are allowed, if they created the query
-		if (count(array_intersect($userRoles, array(ROLE_ID_ASSISTANT, ROLE_ID_AUTHOR, ROLE_ID_REVIEWER)))) {
-			$queryDao = DAORegistry::getDAO('QueryDAO');
-			$query = $queryDao->getById($queryId);
-			if ($query && $query->getHeadNote()->getUserId() == $this->_user->getId()) return true;
+		// Assistants, authors and reviewers are allowed, if they created the query
+		if ($this->hasStageRole($query->getStageId(), array(ROLE_ID_ASSISTANT, ROLE_ID_AUTHOR, ROLE_ID_REVIEWER))) {
```

Comment:
should a SUB_EDITOR be able to edit if he/she created the query?

Reply:
Ooh, good catch. Looks like before subeditors were allowed even if they didn't create the query. I'll add that back in.